BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -270,12 +271,19 @@ NormalBootPath (
 
   BoardInit (EndOfStages);
 
-  CallBoardNotify = TRUE;
-  if ((GetPayloadId () == UEFI_PAYLOAD_ID_SIGNATURE) && (Dst[0] != 0)) {
+  PayloadId = GetPayloadId ();
```

Comment:
What about adding a field in BuildLoader.py -p option?

Reply:
If it is a build option, it might be easily missed by end user.  For OsLoader, we need to enforce this to avoid some issues.   If we have more payloads need to be handled specially,  then we can consider adding build option.